### PR TITLE
Fixes for #2343

### PR DESCRIPTION
**Pull request by c0g1t8**
This build is an aggregate of changes
- update of _gulpfile.js_ to support _Gulp 4_ thus _Node 10_
- complete upgrade of _AllReady.Integration_ to _Canopy2_ and _Core 2.1_ 
- associated changes to _appveyor.yml_

Further comments in #2343

### CODE REVIEW

**Comment by c0g1t8** [comment]
Well it looks like this works.  I'd like some feedback before cleaning up this PR.  Anyone???

File: AllReadyApp/AllReady.IntegrationTest/AllReady.ScenarioTest.fsproj [hunk 1]
```diff
@@ -1,111 +1,22 @@
...
-    </Otherwise>
-  </Choose>
-  <Import Project="$(FSharpTargetsPath)" />
   <ItemGroup>
-    <Compile Include="AssemblyInfo.fs" />
     <Compile Include="Constants.fs" />
+    <Compile Include="Actions.fs" />
     <Compile Include="Pages.fs" />
     <Compile Include="Utils.fs" />
-    <Compile Include="Dates.fs" />
```
Comment (jonparker) [inline AllReadyApp/AllReady.IntegrationTest/AllReady.ScenarioTest.fsproj]: Is Dates.fs no longer needed?

File: AllReadyApp/AllReady.IntegrationTest/Dates.fs [hunk 2]
```diff
@@ -1,17 +0,0 @@
-﻿module Dates
```
Comment (jonparker) [inline AllReadyApp/AllReady.IntegrationTest/Dates.fs]: Ah I see you deleted it.

File: AllReadyApp/Web-App/AllReady/tsconfig.json [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
-    "bin/output"
-  ]
+    "compileOnSave": true,
+    "compilerOptions": {
+        "module": "system",
+        "noEmitOnError": true,
+        "noImplicitAny": false,
+        "removeComments": false,
+        "sourceMap": true,
+        "target": "es5"
```
Comment (jonparker) [inline AllReadyApp/Web-App/AllReady/tsconfig.json]: Updating to ex2018 will be another task!

File: AllReadyApp/AllReady.IntegrationTest/AllReady.ScenarioTest.fsproj (hunk 1 above)
Comment (c0g1t8) [reply AllReadyApp/AllReady.IntegrationTest/AllReady.ScenarioTest.fsproj]: Yes, it was never used.

File: AllReadyApp/AllReady.IntegrationTest/Dates.fs (hunk 2 above)
Comment (c0g1t8) [reply AllReadyApp/AllReady.IntegrationTest/Dates.fs]: It was never used.

File: AllReadyApp/Web-App/AllReady/tsconfig.json (hunk 3 above)
Comment (c0g1t8) [reply AllReadyApp/Web-App/AllReady/tsconfig.json]: Typo :) - I'm pretty sure you mean ES2018.  I'm definitely not up to date with JavaScript and would leave that to others to decide. 

There are so many things related to the client side that needs to discussed:
- possibly moving from _Knockout_ to _Vue_ 
- using _libman_ for package management

**Pull request closed without merging**